flow-typed/box-annotations.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @flow
+ * @file Flow types
+ * @author Box
+ */
+/* eslint-disable no-use-before-define */
+type AnnotationPermissions = {
+    can_edit: boolean,
+    can_delete: boolean
+};
+
+type FilePermissions = {
+    can_annotate: boolean,
+    can_view_annotations_all: boolean,
+    can_view_annotations_self: boolean
+};
+
+type AnnotationType = 'point' | 'plain-highlight' | 'highlight-comment' | 'draw';
+
+type Coordinates = {
+    x: Number,
+    y: Number
+};
+
+// [x1, x2, x3, x4, y1, y2, y3 ,y4]
+type QuadPoint = {
+    x1: Number,
+    y1: Number,
+    x2: Number,
+    y2: Number,
+    x3: Number,
+    y3: Number,
+    x4: Number,
+    y4: Number
+};
+type QuadPoints = Array<QuadPoint>;
+
+type Path = {
+    path: Array<Coordinates>
+};
+type DrawingPaths = Array<Path>;
+
+type LocationInfo = {
+    dimensions: { // original file dimensions
+        x: Number,
+        y: Number
+    },
+    page: Number
+};
+
+type PointLocationInfo = Coordinates & LocationInfo;
+
+type HighlightLocationInfo = {
+    quadpoints: QuadPoints
+} & LocationInfo;
+
+type DrawingLocationInfo = {
+    minX: Number,
+    minY: Number,
+    maxX: Number, 
+    maxY: Number,
+    paths: DrawingPaths
+} & LocationInfo;
+
+type User = {
+    type: 'user',
+    id: string,
+    name: string,
+    login: string,
+    profile_image: string
+};
+
+type AnnotationDetails = {
+    threadID: string,
+    type: string,
+    location: PointLocationInfo | HighlightLocationInfo | DrawingLocationInfo
+};
+
+type BoxFile = {
+    id: string,
+    type: 'file_version'
+};
+
+type Annotation = {
+    id: string, 
+    item: BoxFile,
+    message: string,
+    thread: string,
+    created_by: User,
+    permissions: AnnotationPermissions,
+    created_at: string,
+    modified_at: string,
+    details: AnnotationDetails
```

Comment:
Is this concept of `details` coming from the server? It seems like it should be flattened/normalized before being passed to the components.

Reply:
details is returned from the api

Sample annotation returned from the API
```
{
    created_at:"2018-01-30T17:01:45-08:00",
    created_by: {type: "user", id: "123abc", name: "Billy Bob", login: "email@gmail.com",…},
    details:{
        location:{x: "183.4449", y: "376.1977", page: 2, dimensions: {x: 814.96577946768, y: 1055.9391634981}},
        threadID:"alpha-numeric-id",
        type:"point"
    },
    id:"2431707",
    item:{type: "file_version", id: "456def"},
    message:"I can now continuously create annotations",
    modified_at:"2018-01-30T17:01:45-08:00",
    permissions:{can_edit: true, can_delete: true},
    thread:"1",
    type:"annotation",
}
```

Reply:
Locations look like this: https://github.com/box/box-annotations/pull/211#issuecomment-417034813

Reply:
Anything in details is something that can be modified but just needs to be backwards compatible.